### ISSUE TITLE
test

### ISSUE THREAD

**Issue by fortitude94deng** [4uiiurz1/pytorch-adacos#7]
little brother ,can u tell me how to call metric adacos when testing

**Comment by 4uiiurz1** [comment]
If test classes are same as training classes:
```python
feature = model(input)
output = metric_fc(feature)
```
Otherwise:
```python
master_features = model(master_images)
master_features = F.normalize(master_features)

query_feature = model(query_image)
query_feature = F.normalize(query_feature)
output = F.linear(query_feature, master_features)
```

**Comment by fortitude94deng** [comment]
The full connection layer parameters cannot be saved, the trained model cannot load the full connection layer, can not be predicted, or the prediction is wrong.

**Comment by 4uiiurz1** [comment]
I don't know exactly what you are saying, but I think you should save the model_fc weights as follows:
```python
 torch.save(model_fc.state_dict(), 'model_fc.pth')
```

**Comment by fortitude94deng** [comment]
I mean, your program doesn't save the parameters of the full connection layer. If only the feature layer can be loaded at the time of loading, then the full connection at the back is equivalent to guessing.

**Comment by 4uiiurz1** [comment]
Full connection layer means AdaCos layer?

**Comment by fortitude94deng** [comment]
Yes, I want to know how to save it. Thank you. I will study English well and try to make it clear to you

**Comment by 4uiiurz1** [comment]
I'm not good at English too.
If you save AdaCos layer, add the following code:
```
 torch.save(model_fc.state_dict(), 'model_fc.pth')
```
The model's features have enough representations to identify.
The weights of AdaCos layer are just features representing each class.

**Comment by fortitude94deng** [comment]
：
Well, I understand that. I'm not right after the test, and then I'll ask you, I'll save the full connection layer, thank you.
